secp256k1/keys.go

```diff
@@ -153,28 +156,31 @@ func (s *Signature) UnmarshalJSON(in []byte) error {
 }
 
 func GenerateKeypair() *Keypair {
-	priv := secp256k1.RandomFn()
-	pub := secp256k1.NewPointInfinity()
-	pub.BaseExp(&priv)
-
-	return &Keypair{
-		public: &PublicKey{
-			key: &pub,
-		},
-		private: &PrivateKey{
-			key: &priv,
-		},
+	for {
+		if priv, err := secp256k1.GeneratePrivateKey(); err == nil {
+			pub := new(Point)
+			pub.BaseExp(&priv.Key)
+
+			return &Keypair{
+				public: &PublicKey{
+					key: pub,
+				},
+				private: &PrivateKey{
+					key: &priv.Key,
+				},
+			}
+		}
 	}
```

Comment:
what's the reason for the for loop? in what case would the error be returned?

Reply:
The error is due to a fail reading from `rand.Reader`. On the second thought, in case there's something wrong with reading from `/dev/urandom` immediately trying again won't help, better panic here.